overlord/hookstate/ctlcmd/ctlcmd.go

```diff
@@ -66,7 +71,16 @@ func (c *baseCommand) context() *hookstate.Context {
 	return c.c
 }
 
+func (c *baseCommand) ensureContext() (context *hookstate.Context, err error) {
+	if c.c == nil {
+		err = fmt.Errorf(`The "snapctl %s" command can only be invoked from within a snap`, c.name)
```

Comment:
the slight problem with this new error is that it makes almost sound like there are snapctl commands that can be invoked without a snap, but there are none.

The old messages weren't good but din't have this problem

Reply:
I would use:

"cannot invoke snapctl operation commands (here %q) from outside of a snap", c.name

Reply:
Updated.